app/models/conversation.rb

```diff
@@ -219,6 +219,10 @@ def attempt_confirmation
     result = 'waiting_confirmation'
     if self.ride_confirmed.nil? && self.user
       body = I18n.t(:confirm_ride, locale: user_language, time: ride.pickup_in_time_zone.strftime('%l:%M %P'))
+      
+      # Send email confirmation first.
+      UserMailer.notify_scheduled_ride(user, ride).deliver_now
```

Comment:
What happens if `deliver_now` fails? I'm not asking if actual mail delivery fails. I'm asking about atomicity.

Reply:
I actually didn't mean atomicity among notification methods. I meant atomicity between the database change and the notifications. That is, it would be possible to get into a state where the admins think the ride is confirmed, but no notifications went to the rider.

Reply:
Gotcha. I think we're mixing up the legacy stuff from the bot interactions when someone is scheduling a ride via SMS, when we do need them, as a final step, to confirm they want a ride, with sending out a notification when a scheduled ride is about to happen--part of which also involved confirming the ride.

I moved UserMailer.notify_scheduled_ride up to the calling method, confirm_scheduled_rides, where it should have been in the first place. It only sends email if the voter has one, and it should be fire and forget--the email is opt-out, not opt-in, so the ride should be scheduled regardless.